### DISCUSSION TITLE
Problem with prisma .upsert, Unkown argument

### DISCUSSION THREAD

**Discussion by robokonk** [prisma/prisma#12880] [Q&A]
I have problem with prisma upsert(), I get info:

> PrismaClientValidationError:  Invalid `prisma.prismaUser.upsert()`
> invocation:
> 
> {   where: {
>     email: 'viola@prisma.io'
>     ~~~~~   },   update: {
>     name: 'Viola the Magnificent'   },   create: {
>     email: 'viola@prisma.io',
>     name: 'Viola the Magnificent',
>     profileViews: 0,
>     role: 'admin'   } }
> 
> Unknown arg `email` in where.email for type
> prismaUserWhereUniqueInput. Did you mean `id`? Available args: type
> prismaUserWhereUniqueInput {   id?

My code:
schema.prisma
```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model prismaUser {
  id           Int    @id @default(autoincrement())
  name         String @db.VarChar(255)
  email        String @db.VarChar(255)
  profileViews Int
  role         String @db.VarChar(255)
}
```

node.js
```
const { PrismaClient } = require("@prisma/client");

const prisma = new PrismaClient();

// A `main` function so that you can use async/await
async function main() {
  await prisma.prismaUser.upsert({
    where: {
      email: "viola@prisma.io",
    },
    update: {
      name: "Viola the Magnificent",
    },
    create: {
      email: "viola@prisma.io",
      name: "Viola the Magnificent",
      profileViews: 0,
      role: "admin",
    },
  });
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

Anyone can help me and explain what is wrong?

**Comment by austincrim** [comment]
Hey @robokonk 👋 ,

You will need to add the `@unique` attribute on the `email` field.

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model prismaUser {
  id           Int    @id @default(autoincrement())
  name         String @db.VarChar(255)
  email        String @unique @db.VarChar(255)
  profileViews Int
  role         String @db.VarChar(255)
}
```

This adds a unique constraint in the underlying DB and allows Prisma to use the field to uniquely identify a row for operations like `findUnique`, `delete`, and `upsert`.

Let me know if I can clarify anything further!

**Reply by robokonk** [reply]
@austincrim thanks, but email is not unique records, it is okay?
I have this same email in 5 rows, upsert only working with unique rows?

Prisma support similar method for not unique rows, as uppsertAll?

**Reply by austincrim** [reply]
We do not currently support an `upsertMany` method, but we have an [outstanding feature request](https://github.com/prisma/prisma/issues/4134). Adding +1 and unique usecase in that issue would be really helpful. In the meantime, that issue contains a few workarounds that other users have found.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
